internal/attestation/measurements/fetchmeasurements.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+package measurements
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/edgelesssys/constellation/v2/internal/api/versionsapi"
+	"github.com/edgelesssys/constellation/v2/internal/attestation/variant"
+	"github.com/edgelesssys/constellation/v2/internal/cloud/cloudprovider"
+	"github.com/edgelesssys/constellation/v2/internal/sigstore"
+	"github.com/edgelesssys/constellation/v2/internal/sigstore/keyselect"
+)
+
+// ErrRekor is returned when verifying measurements with Rekor fails.
+var ErrRekor = errors.New("verifying measurements with Rekor")
```

Comment:
```suggestion
// RekorError is returned when verifying measurements with Rekor fails.
type RekorError struct {
	err error
}

// Error returns the error message
func (e *RekorError) Error() string {
	return fmt.Sprintf("verifying measurements with Rekor failed: %s", e.err)
}

// Unwrap returns the wrapped error
func (e *RekorError) Unwrap() error {
	return e.err
}
```

From your code it seems like this is what you actually want to use.
To create the error:
```Go
if err := sigstore.VerifyWithRekor(ctx, publicKey, m.rekor, hash); err != nil {
	return nil, &RekorError{err)
}
```

To check for the error type, e.g. in `cli/internal/cmd/configfetchmeasurements.go`:

```Go
_, err := verifier.FetchAndVerifyMeasurements(...)
var rekErr *RekorError
if errors.As(err, &rekErr) {
	// handle error
}
```